### PR TITLE
chore/fix(EMS-3291): No PDF - Change your answers - Business - Alternative trading address 

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:
This PR adds E2E test coverage for changing the `TRADING_ADDRESS` answer from "yes" to "no" and vice versa.

Also, fixes an issue where, when changing `TRADING_ADDRESS` from "yes" to "no" and then back to "yes", the originally submitted address would be retained.

## Resolution :heavy_check_mark:
- Add E2E tests for changing the `TRADING_ADDRESS` answer from "yes" to "no" via:
  - "Application - Check your answers - Business".
  - "Application - Business - Check your answers".
- Update `mapAndSave.companyDetails` to conditionally nullify different trading address data via `save.companyDifferentTradingAddress`.
- Remove `stripEmptyFormFields` from `save.companyDifferentTradingAddress`.
- Create new helper functions:
  - `nullifyCompanyDifferentTradingAddress`.
  - `shouldNullifyCompanyDifferentAddress`.

## Miscellaneous :heavy_plus_sign:
- Simplify some E2E tests.
- Rename `ApplicationDifferentTradingAddress` type to `ApplicationCompanyDifferentTradingAddress`

### CODE REVIEW

File: src/ui/server/controllers/insurance/business/map-and-save/company-details/index.ts [hunk 1]
```diff
@@ -14,20 +17,51 @@ import mapSubmittedData from '../../map-submitted-data/company-details';
...
+       */
+      const { hasDifferentTradingAddress } = formBody;
+
+      const {
+        company: {
+          differentTradingAddress: { fullAddress },
+        },
+      } = application;
+
+      if (shouldNullifyCompanyDifferentAddress(hasDifferentTradingAddress, fullAddress)) {
```
Comment (Zainzzkk) [inline src/ui/server/controllers/insurance/business/map-and-save/company-details/index.ts:53]: should we put this in a helper and return the saveResponse from that helper?

Comment (ttbarnes) [reply src/ui/server/controllers/insurance/business/map-and-save/company-details/index.ts:52]: Good shout, moved into a new function.

File: src/ui/server/controllers/insurance/business/map-and-save/company-details/index.ts [hunk 2]
```diff
@@ -51,15 +50,7 @@ const companyDetails = async (formBody: RequestBody, application: Application, v
...
-        console.info('Mapping and saving application - business - company details - nullifying different trading address data');
-
-        const nullifiedData = nullifyCompanyDifferentTradingAddress();
-
-        saveResponse = await saveAddress.companyDifferentTradingAddress(application, nullifiedData);
-
-        if (!saveResponse) {
-          return false;
-        }
+        await nullify.companyDifferentTradingAddress(application);
```
Comment (Zainzzkk) [inline src/ui/server/controllers/insurance/business/map-and-save/company-details/index.ts:53]: should we be checking the return value of this incase it's false so we can return false? as in `const response = await nullify.companyDifferentTradingAddress(application);` if !response return false

Comment (ttbarnes) [reply src/ui/server/controllers/insurance/business/map-and-save/company-details/index.ts]: Good shout - updated, also updated for "Loss payee".

**Review by Zainzzkk — APPROVED** [review]

**Pull request merged**